node/node.go

```diff
@@ -415,10 +422,39 @@ func (n *Node) OnStart() error {
 
 	// Always connect to persistent peers
 	if n.config.P2P.PersistentPeers != "" {
-		err = n.sw.DialPeersAsync(n.addrBook, strings.Split(n.config.P2P.PersistentPeers, ","), true)
+		// are any of the persistent peers private?
+		persistentPeers := []string{}
+		persistentAndPrivatePeers := []string{}
+		var privatePeerIDs []string
+		if n.config.P2P.PrivatePeerIDs != "" {
+			privatePeerIDs = strings.Split(n.config.P2P.PrivatePeerIDs, ",")
+		}
+	PP_LOOP:
+		for _, peer := range strings.Split(n.config.P2P.PersistentPeers, ",") {
+			spl := strings.Split(peer, "@")
+			if len(spl) == 2 {
+				for _, ppID := range privatePeerIDs {
+					if spl[0] == ppID {
+						persistentAndPrivatePeers = append(persistentAndPrivatePeers, peer)
+						continue PP_LOOP
+					}
+				}
+			}
+			persistentPeers = append(persistentPeers, peer)
+		}
+
+		err = n.sw.DialPeersAsync(n.addrBook, persistentPeers, true)
 		if err != nil {
 			return err
 		}
+
+		// if any of the persistent peers are private, do not add them to addrbook
+		if len(persistentAndPrivatePeers) > 0 {
+			err = n.sw.DialPeersAsync(nil, persistentAndPrivatePeers, true)
```

Comment:
Private Peers are in general validators that we never dial, so doesn't really make sense for it to be persistent (we will fail to ever re-dial them ...).

I don't think there shoudl ever be overlap between persistent and private peers, right ? Unless folks do want to be able to dial their validators directly... hm

Reply:
right, but someone at some point will add persistent peer to private for whatever reason and will create an issue saying "private peers feature does not work"

Reply:
ok, I'll remove this code and leave a comment